lib/plugins/model.js

```diff
@@ -204,13 +206,14 @@ function findModelDefinitions(rootDir, sources) {
   return registry;
 }
 
-function loadModelDefinition(rootDir, jsonFile, allFiles) {
+function loadModelDefinition(rootDir, jsonFile, allFiles, scriptExtensions) {
   var definition = require(jsonFile);
   var basename = path.basename(jsonFile, path.extname(jsonFile));
   definition.name = definition.name || _.upperFirst(_.camelCase(basename));
 
   // find a matching file with a supported extension like `.js` or `.coffee`
-  var sourceFile = fixFileExtension(jsonFile, allFiles, true);
+  var sourceFile = fixFileExtension(jsonFile, allFiles, true, scriptExtensions);
```

Comment:
@clarkorz In Jest, we will provide `options.scriptExtensions = ['.js']` and it will return when it reaches [`if (isPreferredExtension(filepath, scriptExtensions)) return filepath;`](https://github.com/strongloop/loopback-boot/pull/240/files/acef509e72bafaea6c809a28ff44739871f15e72#diff-b2a704f92515ae553ab94d792a7b7e8c) before getting into that condition.